Packages/com.unity.inputsystem/Tests/InputSystem/Plugins/XInputTests.cs

```diff
@@ -127,5 +130,190 @@ public void Devices_SupportsControllerOnXbox()
         AssertButtonPress(gamepad, new XboxOneGamepadState().WithButton(XboxOneGamepadState.Button.Paddle4), gamepad.paddle4);
     }
 
+#endif
+
+#if UNITY_STANDALONE_OSX || UNITY_EDITOR_OSX
+
+    // Going to simulate the report that comes from
+    // https://github.com/360Controller/360Controller/blob/master/360Controller/ControlStruct.h
+    [StructLayout(LayoutKind.Explicit)]
+    struct MICEXboxControllerForOSXState : IInputStateTypeInfo
+    {
+        [FieldOffset(0)] public byte command;
+        [FieldOffset(1)] public byte size;
+        [FieldOffset(2)] public short buttons;
+        [FieldOffset(4)] public byte triggerLeft;
+        [FieldOffset(5)] public byte triggerRight;
+        [FieldOffset(6)] public short leftX;
+        [FieldOffset(8)] public short leftY;
+        [FieldOffset(10)] public short rightX;
+        [FieldOffset(12)] public short rightY;
+
+        public FourCC GetFormat()
+        {
+            return new FourCC('H', 'I', 'D');
+        }
+    }
```

Comment:
Let's move the structs into the XInput plugin next to replace the current JSON layouts.